library/src/main/java/com/chuckerteam/chucker/internal/support/JsonConverter.kt

```diff
@@ -1,19 +1,14 @@
 package com.chuckerteam.chucker.internal.support
 
-import com.google.gson.FieldNamingPolicy
-import com.google.gson.Gson
-import com.google.gson.GsonBuilder
-import com.google.gson.internal.bind.DateTypeAdapter
-import java.util.Date
+import com.squareup.moshi.Moshi
+import com.squareup.moshi.kotlin.reflect.KotlinJsonAdapterFactory
 
 internal object JsonConverter {
 
-    val instance: Gson by lazy {
-        GsonBuilder()
-            .serializeNulls()
-            .setPrettyPrinting()
-            .setFieldNamingPolicy(FieldNamingPolicy.LOWER_CASE_WITH_UNDERSCORES)
-            .registerTypeAdapter(Date::class.java, DateTypeAdapter())
-            .create()
+    val instance: Moshi by lazy {
+        Moshi.Builder()
+            .add(DateJsonAdapter())
+            .add(KotlinJsonAdapterFactory())
```

Comment:
With this line you are killing one of advantages mentioned in the Reddit thread you shared - size.
If you read the documentation for Moshi carefully, you could find such line:
`Note that the reflection adapter transitively depends on the kotlin-reflect library which is a 2.5 MiB .jar file.`
Info taken from here: https://github.com/square/moshi#reflection

Reply:
So the idea is to replace this approach to code generation. Only with code generation we will get the advantage of Moshi of avoiding reflection and won't have to add `kotlin-reflect` into the library.
Here is the part of documentation about code generation:
https://github.com/square/moshi#codegen

Reply:
Replaced `moshi-kotlin` with `moshi-kotlin-codegen`.